src/main/java/io/github/moulberry/notenoughupdates/profileviewer/SacksPage.kt

```diff
@@ -0,0 +1,505 @@
+            (guiTop + 91).toFloat(),
+            102
+        )
+
+        Utils.renderAlignedString(
+            "§2Items",
+            "§f${StringUtils.formatNumber(currentSackData.itemCount)}",
+            (guiLeft + 27).toFloat(),
+            (guiTop + 108).toFloat(),
+            102
+        )
+
+        GlStateManager.enableDepth()
+
+        var xIndex = 0
+        var yIndex = 0
+
+        val sackTypes = sacksJson.getAsJsonObject("sacks")
+
+        val startIndex = page * pageSize
+        val endIndex = (page + 1) * pageSize
+        var itemCount = -1
+        if (currentSack == "All") {
+            for ((sackName, sackData) in sackTypes.entrySet()) {
+                itemCount++
+                if (itemCount < startIndex || itemCount >= endIndex) continue
+
+                if (yIndex == rows) continue
+                val data = sackData.asJsonObject
+
+                if (!data.has("item") || !data.get("item").isJsonPrimitive || !data.get("item").asJsonPrimitive.isString) continue
+                val sackItemName = data.get("item").asString
+                val itemStack = manager.createItem(sackItemName)
+
+                val x = guiLeft + 168 + xIndex * 37
+                val y = guiTop + 20 + yIndex * 41
+
+                MC.textureManager.bindTexture(GuiProfileViewer.pv_elements)
+                Utils.drawTexturedRect(
+                    (x).toFloat(),
+                    (y).toFloat(),
+                    20f,
+                    20f,
+                    0f,
+                    20 / 256f,
+                    0f,
+                    20 / 256f,
+                    GL11.GL_NEAREST
+                )
+
+                val sackInfo = sackContents[sackName] ?: SackInfo(0, 0.0)
+                Utils.drawStringCentered(
+                    "§6${StringUtils.shortNumberFormat(sackInfo.sackValue.roundToDecimals(0))}",
+                    x + 10,
+                    y - 4,
+                    true,
+                    0
+                )
+                Utils.drawStringCentered(
+                    "§2${StringUtils.shortNumberFormat(sackInfo.itemCount)}",
+                    x + 10,
+                    y + 26,
+                    true,
+                    0
+                )
+                GlStateManager.color(1f, 1f, 1f, 1f)
+
+                if (itemStack != null) {
+                    Utils.drawItemStack(itemStack, x + 2, y + 2)
+
+                    if (mouseX > x && mouseX < x + 20) {
+                        if (mouseY > y && mouseY < y + 20) {
+                            tooltipToDisplay = createTooltip("$sackName Sack", sackInfo.sackValue, sackInfo.itemCount)
+                        }
+                    }
+                } else {
+                    println("$sackItemName missing in neu repo")
+                }
+
+                xIndex++
+                if (xIndex == columns) {
+                    xIndex = 0
+                    yIndex++
+                }
```

Comment:
This feels like we are duplicating state. Can't both of those be calculated using the index instead?
```kt
val xIndex = index % columns
val yIndex = index / columns
if (yIndex >= rows) { break }
```